raiden/tests/integration/network/transport/rif_comms/test_client.py

```diff
@@ -90,123 +87,122 @@ def test_connect_peers():
+    assert client_1._is_subscribed_to(client_2.rsk_address.address) is True
+    assert client_2._is_subscribed_to(client_1.rsk_address.address) is False
+    assert client_2._is_subscribed_to(client_2.rsk_address.address) is False
 
-    # subscribe from node 2 to 1 and check subscriptions
-    topic_id_2, _ = client_2.subscribe_to(address_1)
+    topic_id_2, _ = client_2.subscribe_to(client_1.rsk_address.address)
     assert topic_id_2
-    assert client_1._is_subscribed_to(address_1) is False
-    assert client_1._is_subscribed_to(address_2) is True
-    assert client_2._is_subscribed_to(address_1) is True
-    assert client_2._is_subscribed_to(address_2) is False
+    assert client_1._is_subscribed_to(client_1.rsk_address.address) is False
+    assert client_1._is_subscribed_to(client_2.rsk_address.address) is True
+    assert client_2._is_subscribed_to(client_1.rsk_address.address) is True
+    assert client_2._is_subscribed_to(client_2.rsk_address.address) is False
 
 
-@pytest.mark.parametrize("amount_of_nodes", [2])
-def test_subscribe_to_repeated(comms_nodes):
-    comms_node_1, comms_node_2 = comms_nodes[1], comms_nodes[2]
-    client_1, address_1 = comms_node_1.client, comms_node_1.address
-    client_2, address_2 = comms_node_2.client, comms_node_2.address
+@pytest.mark.parametrize("nodes_to_clients", [{"A": 1, "B": 1}, {"A": 2}])
+def test_subscribe_to_repeated(comms_clients):
+    client_1 = comms_clients[1]
+    client_2 = comms_clients[2]
 
     # subscribe from node 1 to 2 and check subscription
-    topic_id_1, _ = client_1.subscribe_to(address_2)
+    topic_id_1, _ = client_1.subscribe_to(client_2.rsk_address.address)
     assert topic_id_1
-    assert client_1._is_subscribed_to(address_2) is True
+    assert client_1._is_subscribed_to(client_2.rsk_address.address) is True
 
     # subscribe again and check subscription
-    topic_id_2, _ = client_1.subscribe_to(address_2)
-    assert topic_id_2 == topic_id_1
-    assert client_1._is_subscribed_to(address_2) is True
+    topic_id_2, _ = client_1.subscribe_to(client_2.rsk_address.address)
+    assert topic_id_1 == topic_id_2
+    assert client_1._is_subscribed_to(client_2.rsk_address.address) is True
 
 
-@pytest.mark.parametrize("amount_of_nodes", [1])
+@pytest.mark.parametrize("nodes_to_clients", [{"A": 1}])
 @pytest.mark.skip(reason="hangs, incomplete")
-def test_send_message_invalid(comms_nodes):
-    client, address = comms_nodes[1].client, comms_nodes[1].address
-
+def test_send_message_invalid(comms_clients):
+    client = comms_clients[1]
     # send message to unregistered peer
-    # FIXME: hangs
+    # FIXME: raises no peers on routing table
     client.send_message("echo", generate_address())
 
 
-@pytest.mark.parametrize("amount_of_nodes", [1])
-def test_send_message_self(comms_nodes):
-    client, address = comms_nodes[1].client, comms_nodes[1].address
+@pytest.mark.parametrize("nodes_to_clients", [{"A": 1}])
+def test_send_message_self(comms_clients):
```

Comment:
so, when subscribing to/unsubscribing from _self_, we are usually testing 2 clients on the same node to verify this works OK with the multi address implementation.

something similar could be done to `send_message_self`; e.g. clients 1 and 2 send different messages to themselves, but in the test you check that the payload is as expected (and they are not receiving the messages from the other client instead)